src/Control/Coroutine/Aff.purs

```diff
@@ -16,8 +16,25 @@ import Control.Monad.Eff (Eff)
 import Control.Monad.Eff.Class (liftEff)
 import Control.Monad.Free.Trans (hoistFreeT)
 import Control.Monad.Trans.Class (lift)
+import Data.Either (Either(..))
 
-import Data.Either (Either)
+data Emission a b
+  = Step a
+  | Result b
+
+toEither :: forall a b. Emission a b -> Either a b
+toEither = case _ of
+  Step a -> Left a
+  Result b -> Right b
+
+newtype Emitter m a r eff = Emitter (Emission a r -> m (avar :: AVAR | eff) Unit)
```

Comment:
Why not combine `m (avar :: AVAR | eff)` into one type argument here?

Reply:
I was trying to simplify the type on the LHS of the callback as much as possible. It's definitely a little questionable, as "hiding" the row in here could make the lining up of `eff` type variable less obvious, but with the effect row change I guess it won't matter, and it'll just be `m` anyway then.